### PR TITLE
Fix workflows

### PR DESCRIPTION
**Pull request by ivarne**
I noticed that we run lots of pull request actions for renovate PRs when we merge to main, so this will disable a few of the less likely actions to fail for renovate.

* Don't run code analyzis on renovate pull requests (they trigger on every merge to master)
* Use list syntax for branches
* Don't refernece the v8 branch

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -2,10 +2,12 @@ name: "CodeQL"
 
 on:
   push:
-    branches: [ "main", "v8" ]
+    branches: 
+      - "main"
```
Comment (tjololo) [inline .github/workflows/codeql.yml:6]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -2,10 +2,12 @@ name: "CodeQL"
...
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [ "main", "v8" ]
+    branches: 
+      - "main"
```
Comment (tjololo) [inline .github/workflows/codeql.yml:10]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/dotnet-format.yml [hunk 3]
```diff
@@ -2,11 +2,13 @@ name: Verify dotnet format
 
 on:
   pull_request:
-    branches: [ "main", "v8" ]
+    branches: 
+      - "main"
```
Comment (tjololo) [inline .github/workflows/dotnet-format.yml:6]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/dotnet-test.yml [hunk 4]
```diff
@@ -1,9 +1,11 @@
 name: Build and Test on windows, macos and ubuntu
 on:
   push:
-    branches: [ main, v8 ]
+    branches:
+      - "main"
```
Comment (tjololo) [inline .github/workflows/dotnet-test.yml:5]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/dotnet-test.yml [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
   pull_request:
-    branches: [ main, v8 ]
+    branches:
+      - "main"
```
Comment (tjololo) [inline .github/workflows/dotnet-test.yml:8]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/test-and-analyze-fork.yml [hunk 6]
```diff
@@ -1,7 +1,8 @@
 name: Code test and analysis (fork)
 on:
   pull_request:
-    branches: [ main, v8 ]
+    branches:
+      - "main"
```
Comment (tjololo) [inline .github/workflows/test-and-analyze-fork.yml:5]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/test-and-analyze.yml [hunk 7]
```diff
@@ -1,14 +1,17 @@
 name: Code test and analysis
 on:
   push:
-    branches: [ main, v8 ]
+    branches:
+      - "main"
```
Comment (tjololo) [inline .github/workflows/test-and-analyze.yml:5]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/test-and-analyze.yml [hunk 8]
```diff
@@ -1,14 +1,17 @@
...
   pull_request:
-    branches: [ main, v8 ]
+    branches:
+      - "main"
```
Comment (tjololo) [inline .github/workflows/test-and-analyze.yml:8]:
```suggestion
      - "main"
      - "v7"
```

File: .github/workflows/codeql.yml [hunk 9]
```diff
@@ -17,7 +19,8 @@ jobs:
       actions: read
       contents: read
       security-events: write
-
+    if: |
+      github.event.pull_request.user.login != 'renovate[bot]'
```
Comment (tjololo) [inline .github/workflows/codeql.yml]: Could we keep codeQL? there is no reason to why it should fail, but it doesn't hurt to run either IMO

**Review by tjololo — CHANGES_REQUESTED** [review]
Lets add v7 instead of v8.

**Comment by ivarne** [comment]
@tjololo Adding `"v7"` in the `"main"` branch does nothing (it needs to be on the `v7` branch to take effect), so I'm not sure it makes sense to even add them here. 

With regards to CodeQL I have no opinion, but the main goal was to reduce the build storm so I just took everything that wasn't the simple `dotnet test`. As we don't have a warning free main branch (so we can run warning as error) it might make sense to include one of the analysis tools that might detect that an upgrade caused new warnings.

**Comment by tjololo** [comment]
> @tjololo Adding `"v7"` in the `"main"` branch does nothing (it needs to be on the `v7` branch to take effect), so I'm not sure it makes sense to even add them here.
> 
> With regards to CodeQL I have no opinion, but the main goal was to reduce the build storm so I just took everything that wasn't the simple `dotnet test`. As we don't have a warning free main branch (so we can run warning as error) it might make sense to include one of the analysis tools that might detect that an upgrade caused new warnings.

Sorry, you are correct, adding v7 here solves nothing. I will remove the suggestions

File: .github/workflows/codeql.yml (hunk 9 above)
Comment (tjololo) [inline .github/workflows/codeql.yml]:
```suggestion
```

**Review by tjololo — APPROVED** [review]
I suggest keeping one analysis workflow. As codeQL generates no comments if everything is OK I prefer that over sonar

**Comment by ivarne** [comment]
I'll have another look at this tomorrow. I just realized that branch restrictions for pull requests should be removed altogether. It's a shitty heuristic to think that a PR to the payment branch is less likely to benefit from analysis than one to main. Branch PR are so rare anyway.

**Pull request merged**